agrona/src/main/java/org/agrona/concurrent/status/AtomicCounter.java

```diff
@@ -330,27 +435,51 @@ public boolean compareAndSet(final long expectedValue, final long updateValue)
     }
 
     /**
-     * Get the latest value for the counter with volatile semantics.
```

Comment:
I have removed 'latest' because it implies that you will see the latest value. This isn't true. A correctly synchronized program will only show sequential consistent executions, and for sequential consistency, the real time order doesn't need to respected. For that you need linearizability.

The JVM will guarantee visibility, so it will promise that a value is seen eventually (but not immediately).